### PR TITLE
Improve TreeView less styles

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

**Comment by San4es** [comment]
CSS classes renaming, massive changes in LESS hierarchy are looking like a breaking change to me - it can break user CSS customization by class selectors.

Can we:
 - do this changes for 18_2 branch **first**?
 - do this changes for 18_2 branch **only**?

File: js/ui/tree_view/ui.tree_view.base.js [hunk 1]
```diff
@@ -32,11 +32,11 @@ var WIDGET_CLASS = "dx-treeview",
     ITEM_DATA_KEY = "dx-treeview-item-data",
     IS_LEAF = "dx-treeview-node-is-leaf",
     EXPAND_EVENT_NAMESPACE = "dxTreeView_expand",
-    TOGGLE_ITEM_VISIBILITY_CLASS = "dx-treeview-toggle-item-visibility",
```
Comment (mrjohn0011) [inline js/ui/tree_view/ui.tree_view.base.js:35]: We should discuss about the possible breaking change with @jsdmitry and support team

File: js/ui/tree_view/ui.tree_view.base.js [hunk 2]
```diff
@@ -32,11 +32,11 @@ var WIDGET_CLASS = "dx-treeview",
...
+    EXPAND_ICON_CLASS = "dx-treeview-expand-icon",
     LOAD_INDICATOR_CLASS = "dx-treeview-loadindicator",
     LOAD_INDICATOR_WRAPPER_CLASS = "dx-treeview-loadindicator-wrapper",
     NODE_LOAD_INDICATOR_CLASS = "dx-treeview-node-loadindicator",
-    TOGGLE_ITEM_VISIBILITY_OPENED_CLASS = "dx-treeview-toggle-item-visibility-opened",
```
Comment (mrjohn0011) [inline js/ui/tree_view/ui.tree_view.base.js:39]: We should discuss about the possible breaking change with @jsdmitry and support team

File: styles/widgets/android5/menu.android5.less [hunk 3]
```diff
@@ -97,7 +97,7 @@
     }
 
     .dx-treeview {
-        .dx-treeview-toggle-item-visibility {
```
Comment (mrjohn0011) [inline styles/widgets/android5/menu.android5.less:100]: I think this selector can be replaced to the root level

File: styles/widgets/android5/treeView.android5.less [hunk 4]
```diff
@@ -44,99 +44,99 @@
...
+.dx-treeview-item {
+    padding: 8px 6px 6px 3px;
+    min-height: 42px;
 
-        .dx-treeview-item-content {
-            color: @ANDROID5_TREE_VIEW_TEXT_COLOR;
-        }
-
-        .dx-icons-dark();
+    .dx-treeview-item-content {
```
Comment (mrjohn0011) [inline styles/widgets/android5/treeView.android5.less:107]: I think this selector can be replaced to the root level

File: styles/widgets/common/menu.less [hunk 5]
```diff
@@ -49,7 +49,7 @@
 
 .dx-menu-adaptive-mode {
     .dx-treeview {
```
Comment (mrjohn0011) [inline styles/widgets/common/menu.less:51]: Can we remove .dx-treeview selector?

File: styles/widgets/generic/treeView.generic.less [hunk 6]
```diff
@@ -82,123 +82,124 @@
...
         }
     }
+}
+
+
+.dx-treeview-search {
+    margin-bottom: @GENERIC_TREEVIEW_SEARCHBOX_MARGIN_BOTTOM;
+}
+
+&.dx-treeview-with-search > .dx-scrollable {
```
Comment (mrjohn0011) [inline styles/widgets/generic/treeView.generic.less:103]: Can we remove '&' from the root-level selector?

File: styles/widgets/generic/treeView.generic.less [hunk 7]
```diff
@@ -82,123 +82,124 @@
...
         }
+    }
 
-        &:not(.dx-treeview-item-with-checkbox) {
-            &.dx-state-selected {
-                > .dx-treeview-item {
-                    color: @treeview-item-selected-color;
-                    background-color: @treeview-item-selected-bg;
-                }
+    &:not(.dx-treeview-item-with-checkbox) {
```
Comment (mrjohn0011) [inline styles/widgets/generic/treeView.generic.less:158]: Can we get rid of &:not selector?

**Review by mrjohn0011 — CHANGES_REQUESTED** [review]

File: styles/widgets/generic/treeView.generic.less [hunk 8]
```diff
@@ -82,123 +82,124 @@
...
-                > .dx-treeview-item {
-                    background-color: @treeview-focused-bg;
+        &.dx-state-focused {
+            > .dx-treeview-item {
+                background-color: @treeview-focused-bg;
 
-                    * {
-                        color: @treeview-focus-color;
-                    }
+                * {
```
Comment (mrjohn0011) [inline styles/widgets/generic/treeView.generic.less:170]: Can we remove '*'?

File: styles/widgets/ios7/treeView.ios7.less [hunk 9]
```diff
@@ -27,91 +27,91 @@
...
         }
     }
 
-    .dx-treeview-item {
-        padding: 8px 6px 6px 6px;
-        min-height: 32px;
-
-        .dx-icon {
-            .dx-icon-sizing(24px);
+    &:not(.dx-treeview-item-with-checkbox) {
```
Comment (mrjohn0011) [inline styles/widgets/ios7/treeView.ios7.less:79]: Try to simplify selectors with 'background-color' property

File: styles/widgets/android5/treeView.android5.less [hunk 10]
```diff
@@ -44,99 +44,99 @@
...
-    &.dx-treeview-with-search > .dx-scrollable {
-        height: calc(~"100% - "@ANDROID5_TREEVIEW_SEARCH_EDITOR_HEIGHT);
+    &.dx-rtl {
+        .dx-treeview-make-rtl(@CONTAINER_WITH_BORDER_OFFSET, @NODE_OFFSET, @ITEM_WITH_CHECKBOX_OFFSET, @CHECKBOX_OFFSET, @SELECT_ALL_TEXT_OFFSET, @SELECT_ALL_ITEM_LEFT_PADDING, @BORDER_PADDING);
     }
+}
 
-    &.dx-treeview-border-visible {
-        border: 1px solid @ANDROID5_TREE_VIEW_BORDER_COLOR;
+&.dx-treeview-with-search > .dx-scrollable {
```
Comment (DokaRus) [inline styles/widgets/android5/treeView.android5.less:52]: Looks like there is no need to use `&` after a style extracting

File: styles/widgets/android5/treeView.android5.less [hunk 11]
```diff
@@ -44,99 +44,99 @@
...
+    height: calc(~"100% - "@ANDROID5_TREEVIEW_SEARCH_EDITOR_HEIGHT);
+}
 
-        .dx-treeview-select-all-item {
-            padding-left: @SELECT_ALL_ITEM_LEFT_PADDING;
-        }
+&.dx-treeview-border-visible {
+    border: 1px solid @ANDROID5_TREE_VIEW_BORDER_COLOR;
```
Comment (DokaRus) [inline styles/widgets/android5/treeView.android5.less:57]: Looks like there is no need to use `&` after a style extracting

**Comment by DokaRus** [comment]
You need to rebase the PR branch onto 18_2

**Pull request closed without merging**